### DISCUSSION TITLE
Why doesn't createUserWithEmailAndPassword return a promise?

### DISCUSSION THREAD

**Discussion by birdwell** [invertase/react-native-firebase#5209] [Q&A]
I'm calling the `auth().createUserWithEmailAndPassword` but nothing is returned. The user is created in firebase and the auth state changed call happens. However, the promise is never sent back after calling this function. 

```javascript
try {
      const user = await auth().createUserWithEmailAndPassword(email, password);
      console.log(user);
    } catch (error) {
      console.error(error);
    }
```

**Comment by mikehardy** [comment]
It does in my app

```typescript
        const userCredential = await firebase
          .auth()
          .createUserWithEmailAndPassword(email, password);
        // If you need to do something with the user, do it here
        // RootView::userChangedListener will automatically route and unmount this component
        console.log('Register::_onRegisterClick - created the user successfully?', userCredential);
```

**Reply by birdwell** [reply]
Hi @mikehardy, thanks for the quick response. 😄  Is there a difference in firebase.auth() vs auth() directly? I might have set it up incorrectly with ios. It does create them in firebase however. Also, the docs are gorgeous.

**Reply by mikehardy** [reply]
:thinking: there should not be a difference between the two assuming the imports are correct. Can't take credit for the docs that was Mike@ and Elliot@ Invertase to start, and now mostly community contributions if see you anything off you can PR it and I'll get it in.

I'm not sure why your code isn't working but if you try a clean project and a simple App.js I find that's the best way to test-integrate methods from firebase to start - in a simple playground - you can kick out a simple starter like so https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh with everything in there ready for testing

**Reply by birdwell** [reply]
I rebuilt clean and used your code with firebase.auth() instead of the default export as auth().create and it works! First of all thank for your time and suggestion! We are building a new app and using RNFirebase a bunch. I'll make sure to PR if I see anything. They've been super straight forward docs and helpful with the setup.
